google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -241,6 +254,7 @@ protected ServiceOptions(Class<? extends ServiceFactory<ServiceT, OptionsT>> ser
         getFromServiceLoader(rpcFactoryClass, getDefaultRpcFactory()));
     serviceRpcFactoryClassName = serviceRpcFactory.getClass().getName();
     clock = firstNonNull(builder.clock, Clock.defaultClock());
+    transportOptions = firstNonNull(builder.transportOptions, getDefaultTransportOptions());
```

Comment:
Minor (as it is a much older design decision, not related to your changes). I believe it is better to avoid calling non-private methods from constructors, as it is a bad practice, and potentially leads to premature usage of non-initialized subclasses  (for example http://stackoverflow.com/questions/15327417/is-it-ok-to-call-abstract-method-from-constructor-in-java).

In this case passing the default transport options and default rpc factory as ServiceOptions constructor parameters can fix the problem.